feat: support global prefix to userTaskFilterTransformer

nathansandi opened the pull request:
## Description

- Base implementation for client support the globalPrefix
- Include global prefix over the UserTaskFilterTrasnform
- Ajust newInstance calls to support the new parameter

It will be possible apply integration tests once all the indexes is adapted 

## Related issues

- https://github.com/camunda/camunda/issues/23808

nathansandi committed:
feat: support global prefix to userTaskFilterTransformer

nathansandi committed:
feat: support global prefix to userTaskFilterTransformer

nathansandi committed:
fix: refactor format stringName

fix: remove enum and add component name on template

fix: add component name to TaskTemplate

fix: add the newinstance for test

fix: remove not used dependency

nathansandi committed:
fix: refactor format stringName

fix: remove enum and add component name on template

fix: add component name to TaskTemplate

fix: add the newinstance for test

fix: remove not used dependency

Zelldon commented on search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java:
```diff
@@ -62,10 +62,12 @@ public class SearchClients
-      final DocumentBasedSearchClient searchClient, final boolean isCamundaExporterEnabled) {
+      final DocumentBasedSearchClient searchClient,
+      final boolean isCamundaExporterEnabled,
+      final String globalPrefix) {
```
🔧 I guess we can also just call it prefix. :)

Zelldon commented on search/search-client-query-transformer/src/test/java/io/camunda/search/clients/transformers/sort/AbstractSortTransformerTest.java:
```diff
@@ -16,7 +16,7 @@
 public class AbstractSortTransformerTest {
 
-  private final ServiceTransformers transformers = ServiceTransformers.newInstance(false);
+  private final ServiceTransformers transformers = ServiceTransformers.newInstance(false, null);
```
```suggestion
  private final ServiceTransformers transformers = ServiceTransformers.newInstance(false, "");
```

Zelldon commented on search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/ServiceTransformers.java:
```diff
@@ -96,9 +96,10 @@ public final class ServiceTransformers {
   private ServiceTransformers() {}
 
-  public static ServiceTransformers newInstance(final boolean isCamundaExporterEnabled) {
+  public static ServiceTransformers newInstance(
+      final boolean isCamundaExporterEnabled, final String globalPrefix) {
```
❓ Why we have this boolean `isCamundaExporterEnabled` ?

Zelldon commented on search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java:
```diff
@@ -104,8 +109,17 @@ public SearchQuery toSearchQuery(final UserTaskFilter filter) {
+
+      return List.of(indexName);
     }
+    // TDB - The listview is used when the exporter is not enabled, this will be removed in 8.8
```
Ah now I see. Hm but the Camunda Exporter will be standard right?

nathansandi replied:
In. 8.8 yes 👍 , This will be kept for 8.7

ChrisKujawa replied:
Even for 8.7 we will not have an 8.7 importer. :)

Zelldon commented on search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java:
```diff
@@ -104,8 +109,17 @@ public SearchQuery toSearchQuery(final UserTaskFilter filter) {
     if (isCamundaExporterEnabled) {
-      return List.of("tasklist-task-8.5.0_");
+      final String indexName =
+          String.format(
+              TaskTemplate.FULL_QUALIFIED_INDEX_NAME_PATTERN,
+              TaskTemplate.formatIndexPrefix(globalPrefix),
+              TaskTemplate.COMPONENT_NAME,
+              TaskTemplate.INDEX_NAME,
+              TaskTemplate.INDEX_VERSION);
```
🔧 ❓ Maybe we can introduce a static method on the `TaskTemplate` that allows us to simple use it without constructing the format on our own.

Example:

```
   final String indexName = TaskTemplate.getIndexNameWithPrefix(globalPrefix);
```

In TaskTemplate:

```
  public static String getIndexNameWithPrefix(final String prefix) {
    return new TaskTemplate(prefix, false).getFullQualifiedName();
   }
```

Just as an idea to reuse the `getFullQualifiedName` but we could also have the format in the static again.

ChrisKujawa reviewed (approved):
Looks good to me in general, there are just some smaller hints. :)

nathansandi committed:
fix: apply code review suggestions

- Rename from globalPrefix to prefix
- Change null to empty string on test case
- Addd static method for build naming on user task filter transfmorer

nathansandi committed:
fix: apply code review suggestions

- Rename from globalPrefix to prefix
- Change null to empty string on test case
- Addd static method for build naming on user task filter transfmorer

nathansandi committed:
Merge branch 'add-global-prefix-module' of https://github.com/camunda/camunda into add-global-prefix-module

nathansandi committed:
fix: rename globalPrefix to prefix

fix: rename globalPrefix to prefix

nathansandi committed:
fix: checkstyle

nathansandi committed:
fix: test cases for sort

nathansandi committed:
Merge branch 'main' into add-global-prefix-module

nathansandi committed:
Merge branch 'main' into add-global-prefix-module

nathansandi committed:
fix: sort unit test

nathansandi committed:
fix: apply lint

nathansandi committed:
Merge branch 'main' into add-global-prefix-module

nathansandi committed:
Merge remote-tracking branch 'origin/main' into add-global-prefix-module

nathansandi committed:
fix: disable variables test it
